### ISSUE TITLE
Array::lace crashes on singleton

### ISSUE THREAD

**Issue by telephon** [supercollider/supercollider#3716]
```supercollider
[].lace // works, but returns [nil] (seems not a good solution)
[[]].lace // crashes sclang
```

What I would expect is:

```supercollider
[].lace // => `[]`
[[]].lace // => `[]`
```

The helpfile says:

> Returns a new Array whose elements are interlaced sequences of the elements of the receiver's subcollections, up to size length.

**Comment by telephon** [comment]
Also, if I can write:

`[0, 1, 2, 3].lace(4)`
and
`[[0, 1], [2, 3]].lace`

I would expect that I can write
`[0, 1, 2, 3].lace`

But that fails with an error.

**Comment by miguel-negrao** [comment]
https://github.com/supercollider/supercollider/blob/f0d6be8e9475332fe59575b6bb76ac953ea1763c/lang/LangPrimSource/PyrArrayPrimitives.cpp#L2046

obj3 size is not checked, so a division by zero occurs which causes termination with SIGFPE. obj3 size should be checked.

**Comment by lnihlen** [comment]
Taking a look at this today, hoping to have a PR soon.

**Comment by lnihlen** [comment]
For your review. It's my first PR for SC, I've tried to read the various contributing guidelines but please accept my apologies if I missed some contributing etiquette. 

If you test this PR you will now find that now:

```supercollider
[].lace // => `[]`
``` 
But, unlike your proposal, 

```supercollider
[[]].lace // => `[[]]`
```

This is because the way that the Array::lace function handles other non-array elements in the target array. As you point out above, in the patched code

```supercollider
[1, 2, 3, 4].lace // => `[1, 2, 3, 4]`
```

Which I interpret to mean that non-indexable elements, that is things that can't be interleaved, should interleave atomically as themselves. Since an empty array can't be indexed either, to me that means

```supercollider
[1, 2, [], 4].lace // => `[1, 2, [], 4]`
```

Whereas I take the requirement that the nested empty array case should return only an empty array to mean that

```supercollider
[1, 2, [], 4].lace // => `[]`
```

Was curious to hear your thoughts, however, I could always special-case the nested empty array situation to return a single empty array.

**Comment by jamshark70** [comment]
Ahhhh (or arrrgggghhh)... one of those cases where the lack of a spec causes confusion. (By confusion, I mean that I don't know what the right answer is, either.)

> an empty array can't be indexed either

This is not strictly true.

```
[].at(0)
-> nil
```

A strong case could be made that an empty array in `lace` should be indexed (`at`-ed) just like an array with contents, and the result of `at` embedded into the output array (which result will be `nil`).

By that interpretation:

```
[1, 2, [], 4].lace // => `[1, 2, nil, 4]`
```

Again, there is no clear spec, so I don't know what is the right answer. I tend to think there is a longer precedent (in SC, not in other languages) for `Array:at` to return nil for out-of-bounds indices than there is for empty arrays to be treated as special cases or "non-indexable."

**Comment by lnihlen** [comment]
Ah, good point. I patched the PR, empty subarrays will now lace as nil:

```supercollider
[].lace // => `[]`
[[]].lace // => `[ nil ]`
[1, 2, [], 4].lace // => `[1, 2, nil, 4]`
```

**Comment by telephon** [comment]
@jamshark70 I fully agree with your assessment.

>Ahhhh (or arrrgggghhh)... one of those cases where the lack of a spec causes confusion.

I don't think this is the case really.  The reason is that this is a primitive and we don't have proper message passing there. If we had the same method in sclang, the messages sent to the subarrays/subitems (probably something line `size` and `at`) would have given a better indication of what is actually _implied_ without being explicit in words.

**Comment by snappizz** [comment]
fixed in #3774.
